tutorials/Python/11_OOP/OOP.ipynb

```diff
@@ -0,0 +1,769 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Using a dictionary to represent acetylene could be a good solution:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "acetylene = {\n",
+    "    \"elements\": np.array([1, 6, 6, 1]),\n",
+    "    \"coordinates\": np.array([\n",
+    "        [-1.06, 0.00, 0.00], # H1\n",
+    "        [ 0.00, 0.00, 0.00], # C1\n",
+    "        [ 1.20, 0.00, 0.00], # C2\n",
+    "        [ 2.26, 0.00, 0.00], # H2\n",
+    "    ])\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We now have a single variable representing the acetylene molecule. All the information is stored within that variable and it can be easily and unequivocally accessed. If we want to see the elements (atomic numbers) of the  atoms in acetylene, we can simply print them out:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "output_type": "stream",
+     "name": "stdout",
+     "text": [
+      "[1 6 6 1]\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(acetylene[\"elements\"])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Do you see where we are heading? Objects allow us to tie together data (as we just did with a dictionary) and functionality. By defining and using objects we can create a super slick representation of a molecule, with all the data and all the functionality we need."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In the following sections we will have a first look at the simple concepts of object-oriented programming (OOP) to get an idea of this powerful paradigm."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Molecule Class"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "A _class_ is the definition of an object. We can think of a class as an architectural plan and an object (usually called an _instance_ of the class in this context) as an house built following said plan. With the same plan, you can build multiple houses (create multiple instances), each housing different people (different data). For our example above, a yet to be defined `Molecule` class would represent a generic molecule (by defining the data and the functionality associated to a molecule) while `acetylene` would be an object (instance) of said class, containing the actual data defining `acetylene`. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Creating a new class in Python has a similar syntax to a function definition, with the `class` keyword instead of `def` and without `()` (`()` is optional for a class definition):"
```

Comment:
It's weird, the `def` and `()` isn't rendering properly for me on github, if it works for you locally then it's good enough.